privacyidea/api/validate.py

```diff
@@ -539,3 +540,56 @@ def trigger_challenge():
 
     return send_result(result_obj, details=details)
 
+
+@validate_blueprint.route('/polltransaction', methods=['GET'])
+@prepolicy(mangle, request=request)
+@CheckSubscription(request)
+@prepolicy(api_key_required, request=request)
+def poll_transaction():
+    """
+    Given a mandatory transaction ID, check if any non-expired challenge for this transaction ID
+    has been answered. In this case, return true. If this is not the case, return false.
+    This endpoint also returns false if no challenge with the given transaction ID exists.
+
+    This is mostly useful for out-of-band tokens that should poll this endpoint
+    to determine when to send an authentication request to ``/validate/check``.
+
+    :jsonparam transaction_id: a transaction ID
+    """
+    transaction_id = getParam(request.all_data, "transaction_id", required)
+    # Fetch a list of challenges with the given transaction ID
+    # and determine whether it contains at least one non-expired answered challenge.
+    matching_challenges = get_challenges(transaction_id=transaction_id)
+    answered_challenges = extract_answered_challenges(matching_challenges)
+
+    if answered_challenges:
+        result = True
+        log_challenges = answered_challenges
+    else:
+        result = False
+        log_challenges = matching_challenges
```

Comment:
I just noticed something: In case there *are* challenges with the given transaction ID, but *none* of these challenges have been answered, this will write the serials of all challenges with the given transaction ID to the audit log, including serials of expired challenges. Should we only consider non-expired challenges?

Reply:
You are right, I think we should **not** log all the challenges, that are not relevant.
Or we should only log this, if a valid/answered challenge was found.

Reply:
I guess the question is what information would be useful in the audit log: If there *are* challenges with the given transaction ID, but *none* of them has been answered -- should the audit log contain the serials of the unanswered challenges, and the respective user? i.e. would this be useful information to the helpdesk user?

Reply:
I modified the endpoint to only consider *non-expired* challenges in 7e16c1a.